### PR TITLE
reduce package dependency cycles

### PR DESCRIPTION
**Pull request by turadg**
## Description

Progress on https://github.com/Agoric/agoric-sdk/issues/4645

Cycles detected with `yarn lerna run build`. Graphs drawn with `npx lerna-dependency-graph --outputFormat=png`

*Before*
before

<details>
<summary>12 cycles</summary>

```
lerna WARN ECYCLE @agoric/swingset-vat -> @agoric/notifier -> @agoric/swingset-vat
lerna WARN ECYCLE agoric -> @agoric/cache -> @agoric/vats -> @agoric/inter-protocol -> @agoric/cosmic-swingset -> agoric
lerna WARN ECYCLE @agoric/deploy-script-support -> (nested cycle: agoric -> @agoric/cache -> @agoric/vats -> @agoric/inter-protocol -> @agoric/cosmic-swingset -> agoric) -> @agoric/deploy-script-support
lerna WARN ECYCLE @agoric/governance -> (nested cycle: @agoric/deploy-script-support -> (nested cycle: agoric -> @agoric/cache -> @agoric/vats -> @agoric/inter-protocol -> @agoric/cosmic-swingset -> agoric) -> @agoric/deploy-script-support) -> @agoric/governance
lerna WARN ECYCLE @agoric/pegasus -> (nested cycle: @agoric/governance -> (nested cycle: @agoric/deploy-script-support -> (nested cycle: agoric -> @agoric/cache -> @agoric/vats -> @agoric/inter-protocol -> @agoric/cosmic-swingset -> agoric) -> @agoric/deploy-script-support) -> @agoric/governance) -> @agoric/pegasus
lerna WARN ECYCLE @agoric/smart-wallet -> (nested cycle: @agoric/pegasus -> (nested cycle: @agoric/governance -> (nested cycle: @agoric/deploy-script-support -> (nested cycle: agoric -> @agoric/cache -> @agoric/vats -> @agoric/inter-protocol -> @agoric/cosmic-swingset -> agoric) -> @agoric/deploy-script-support) -> @agoric/governance) -> @agoric/pegasus) -> @agoric/smart-wallet
lerna WARN ECYCLE @agoric/store -> (nested cycle: @agoric/swingset-vat -> @agoric/notifier -> @agoric/swingset-vat) -> @agoric/store
lerna WARN ECYCLE @agoric/vat-data -> (nested cycle: @agoric/store -> (nested cycle: @agoric/swingset-vat -> @agoric/notifier -> @agoric/swingset-vat) -> @agoric/store) -> @agoric/vat-data
lerna WARN ECYCLE @agoric/zoe -> (nested cycle: @agoric/smart-wallet -> (nested cycle: @agoric/pegasus -> (nested cycle: @agoric/governance -> (nested cycle: @agoric/deploy-script-support -> (nested cycle: agoric -> @agoric/cache -> @agoric/vats -> @agoric/inter-protocol -> @agoric/cosmic-swingset -> agoric) -> @agoric/deploy-script-support) -> @agoric/governance) -> @agoric/pegasus) -> @agoric/smart-wallet) -> @agoric/zoe
lerna WARN ECYCLE @agoric/wallet-backend -> (nested cycle: @agoric/zoe -> (nested cycle: @agoric/smart-wallet -> (nested cycle: @agoric/pegasus -> (nested cycle: @agoric/governance -> (nested cycle: @agoric/deploy-script-support -> (nested cycle: agoric -> @agoric/cache -> @agoric/vats -> @agoric/inter-protocol -> @agoric/cosmic-swingset -> agoric) -> @agoric/deploy-script-support) -> @agoric/governance) -> @agoric/pegasus) -> @agoric/smart-wallet) -> @agoric/zoe) -> @agoric/wallet-backend
lerna WARN ECYCLE (nested cycle: @agoric/vat-data -> (nested cycle: @agoric/store -> (nested cycle: @agoric/swingset-vat -> @agoric/notifier -> @agoric/swingset-vat) -> @agoric/store) -> @agoric/vat-data) -> (nested cycle: @agoric/vat-data -> (nested cycle: @agoric/store -> (nested cycle: @agoric/swingset-vat -> @agoric/notifier -> @agoric/swingset-vat) -> @agoric/store) -> @agoric/vat-data)
lerna WARN ECYCLE (nested cycle: @agoric/wallet-backend -> (nested cycle: @agoric/zoe -> (nested cycle: @agoric/smart-wallet -> (nested cycle: @agoric/pegasus -> (nested cycle: @agoric/governance -> (nested cycle: @agoric/deploy-script-support -> (nested cycle: agoric -> @agoric/cache -> @agoric/vats -> @agoric/inter-protocol -> @agoric/cosmic-swingset -> agoric) -> @agoric/deploy-script-support) -> @agoric/governance) -> @agoric/pegasus) -> @agoric/smart-wallet) -> @agoric/zoe) -> @agoric/wallet-backend) -> (nested cycle: @agoric/wallet-backend -> (nested cycle: @agoric/zoe -> (nested cycle: @agoric/smart-wallet -> (nested cycle: @agoric/pegasus -> (nested cycle: @agoric/governance -> (nested cycle: @agoric/deploy-script-support -> (nested cycle: agoric -> @agoric/cache -> @agoric/vats -> @agoric/inter-protocol -> @agoric/cosmic-swingset -> agoric) -> @agoric/deploy-script-support) -> @agoric/governance) -> @agoric/pegasus) -> @agoric/smart-wallet) -> @agoric/zoe) -> @agoric/wallet-backend)
```

</details>

*After*
<details>
<summary>8 cycles</summary>

```
lerna WARN ECYCLE Dependency cycles detected, you should fix these!
lerna WARN ECYCLE @agoric/swingset-vat -> @agoric/notifier -> @agoric/swingset-vat
lerna WARN ECYCLE @agoric/deploy-script-support -> @agoric/vats -> @agoric/inter-protocol -> @agoric/deploy-script-support
lerna WARN ECYCLE @agoric/governance -> (nested cycle: @agoric/deploy-script-support -> @agoric/vats -> @agoric/inter-protocol -> @agoric/deploy-script-support) -> @agoric/governance
lerna WARN ECYCLE @agoric/smart-wallet -> (nested cycle: @agoric/governance -> (nested cycle: @agoric/deploy-script-support -> @agoric/vats -> @agoric/inter-protocol -> @agoric/deploy-script-support) -> @agoric/governance) -> @agoric/smart-wallet
lerna WARN ECYCLE @agoric/store -> (nested cycle: @agoric/swingset-vat -> @agoric/notifier -> @agoric/swingset-vat) -> @agoric/store
lerna WARN ECYCLE @agoric/vat-data -> (nested cycle: @agoric/store -> (nested cycle: @agoric/swingset-vat -> @agoric/notifier -> @agoric/swingset-vat) -> @agoric/store) -> @agoric/vat-data
lerna WARN ECYCLE (nested cycle: @agoric/smart-wallet -> (nested cycle: @agoric/governance -> (nested cycle: @agoric/deploy-script-support -> @agoric/vats -> @agoric/inter-protocol -> @agoric/deploy-script-support) -> @agoric/governance) -> @agoric/smart-wallet) -> (nested cycle: @agoric/smart-wallet -> (nested cycle: @agoric/governance -> (nested cycle: @agoric/deploy-script-support -> @agoric/vats -> @agoric/inter-protocol -> @agoric/deploy-script-support) -> @agoric/governance) -> @agoric/smart-wallet)
lerna WARN ECYCLE (nested cycle: @agoric/vat-data -> (nested cycle: @agoric/store -> (nested cycle: @agoric/swingset-vat -> @agoric/notifier -> @agoric/swingset-vat) -> @agoric/store) -> @agoric/vat-data) -> (nested cycle: @agoric/vat-data -> (nested cycle: @agoric/store -> (nested cycle: @agoric/swingset-vat -> @agoric/notifier -> @agoric/swingset-vat) -> @agoric/store) -> @agoric/vat-data)
```

</details>

after

### Security Considerations

--

### Scaling Considerations

--
### Documentation Considerations

This breaks some exports, but ones that were unlikely to be used outside the repo.

### Testing Considerations

CI

### CODE REVIEW

**Comment by erights** [comment]
batched-deliver.js uses ambient authority (`setTimeout`, `clearTimeout`). Even after we isolate separate packages into separate compartments, LavaMoat-like, in order to do least-authority linkage, by moving batched-deliver.js into `internal`, the `internal` package as a whole would have to be given this ambient authority. This violates POLA as nothing else in `internal` needs this authority.

Can batched-deliver.js be fixed so it no longer needs ambient authority? Please?

File: golang/cosmos/x/swingset/keeper/keeper.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ import (
 )
 
 // Top-level paths for chain storage should remain synchronized with
-// packages/vats/src/chain-storage-paths.js
+// packages/agoric/internal/src/chain-storage-paths.js
```
Comment (michaelfig) [inline golang/cosmos/x/swingset/keeper/keeper.go]:
```suggestion
// packages/internal/src/chain-storage-paths.js
```

File: packages/deploy-script-support/test/unitTests/test-assertOfferResult.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
-import { test } from '@agoric/swingset-vat/tools/prepare-test-env-ava.js';
+import test from 'ava';
+import '@endo/init';
```
Comment (michaelfig) [inline packages/deploy-script-support/test/unitTests/test-assertOfferResult.js:2]: This isn't really the same semantics.  Did you do this to be able to drop the devDependency on swingset?

File: packages/internal/src/lib-chainStorage.js [hunk 3]
```diff
@@ -106,7 +142,7 @@ const makeNullStorageNode = () => {
  * falling back to an inert object with the correct interface (but incomplete
  * behavior) when that is unavailable.
  *
- * @param {ERef<StorageNode?>} storageNodeRef
+ * @param {import('@endo/eventual-send').ERef<StorageNode?>} storageNodeRef
```
Comment (michaelfig) [inline packages/internal/src/lib-chainStorage.js]:
```suggestion
 * @param {import('@endo/far').ERef<StorageNode?>} storageNodeRef
```

**Review by michaelfig — APPROVED** [review]
Looking good.

I would prefer that `@agoric/vats/tools/...` be renamed to `@agoric/internal/tools/...` instead of `@agoric/internal/src/...`, but since we already talked about that, I'm willing to live with it.

File: packages/deploy-script-support/test/unitTests/test-assertOfferResult.js (hunk 2 above)
Comment (turadg) [reply packages/deploy-script-support/test/unitTests/test-assertOfferResult.js:2]: yep. the rest of it wasn't totally necessary.

It affects redactions but there's work scheduled to address that broadly https://github.com/endojs/endo/issues/1235

File: golang/cosmos/x/swingset/keeper/keeper.go (hunk 1 above)
Comment (turadg) [reply golang/cosmos/x/swingset/keeper/keeper.go]: good catch. I'll force push a fix for this and the other suggestions.

**Pull request merged**